include/GafferScene/OptionQuery.h

```diff
@@ -0,0 +1,124 @@
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENE_OPTIONQUERY_H
+#define GAFFERSCENE_OPTIONQUERY_H
+
+#include "GafferScene/Export.h"
+#include "GafferScene/ScenePlug.h"
+#include "GafferScene/TypeIds.h"
+
+#include "Gaffer/ArrayPlug.h"
+#include "Gaffer/ComputeNode.h"
+#include "Gaffer/NameValuePlug.h"
+#include "Gaffer/StringPlug.h"
+#include "Gaffer/TypedObjectPlug.h"
+
+#include <string>
+
+namespace GafferSceneModule
+{
+
+void bindQueries();
+
+}
+
+namespace GafferScene
+{
+
+class GAFFERSCENE_API OptionQuery : public Gaffer::ComputeNode
+{
+	public :
+		OptionQuery( const std::string& name = defaultName< OptionQuery >() );
+		~OptionQuery() override;
+
+		GAFFER_NODE_DECLARE_TYPE( GafferScene::OptionQuery, OptionQueryTypeId, Gaffer::ComputeNode );
+
+		ScenePlug *scenePlug();
+		const ScenePlug *scenePlug() const;
+
+		Gaffer::ArrayPlug *queriesPlug();
+		const Gaffer::ArrayPlug *queriesPlug() const;
+
+		Gaffer::ArrayPlug *outPlug();
+		const Gaffer::ArrayPlug *outPlug() const;
+
+		/// Adds a query for option, with a type and default value specified by plug.
+		/// The returned NameValuePlug is parented to queriesPlug() and may be edited
+		/// subsequently to modify the option name and default. Corresponding children
+		/// are added to existsPlug() and valuePlug() to provide the output from the query.
+		Gaffer::NameValuePlug *addQuery(
+			const Gaffer::ValuePlug *plug,
+			const std::string &option = ""
+		);
+		/// Removes a query. Throws an Exception if the query or corresponding children
+		/// of `valuesPlug()` and `existsPlug()` can not be deleted.
+		void removeQuery( Gaffer::NameValuePlug *plug );
+
+		void affects( const Gaffer::Plug* input, AffectedPlugsContainer& outputs ) const override;
+
+		/// Returns the `exists`, `value` or child of `out` corresponding to the specified
+		/// query plug. Throws an exception if the query does not exist or the corresponding output
+		/// plug does not exist or is the wrong type.
+		const Gaffer::BoolPlug *existsPlugFromQuery( const Gaffer::NameValuePlug *queryPlug ) const;
+		const Gaffer::ValuePlug *valuePlugFromQuery( const Gaffer::NameValuePlug *queryPlug ) const;
+		const Gaffer::ValuePlug *outPlugFromQuery( const Gaffer::NameValuePlug *queryPlug ) const;
+
+		/// Returns the child of `queryPlug` or `outPlug` corresponding to the `outputPlug`.
+		/// `outputPlug` can be any descendant of the desired ancestor.
+		/// Throws an exception if there is no corresponding query or the result is the wrong type.
+		const Gaffer::NameValuePlug *queryPlug( const Gaffer::ValuePlug *outputPlug ) const;
+		const Gaffer::ValuePlug *outPlug( const Gaffer::ValuePlug *outputPlug ) const;
+
+	protected :
+
+		const std::string namePrefix() const;
```

Comment:
I don't think this buys us anything, for the same reasons I described for OptionTweaks. Better to drop it I think...

Reply:
Removed in b30c5421125540f1262f330eb7c18b222fc813dc along with a bit more binding simplification now that we don't have an extra method in `OptionQuery`.